src/mem/alloc.h

```diff
@@ -1476,17 +1503,38 @@ namespace snmalloc
       // a real allocator and construct one if we aren't.
       if (NeedsInitialisation(this))
       {
-        InitThreadAllocator([p](void* alloc) {
-          reinterpret_cast<Allocator*>(alloc)->dealloc(p);
+        InitThreadAllocator([target, p, sizeclass](void* alloc) {
+          reinterpret_cast<Allocator*>(alloc)->dealloc_not_large(
+            target, p, sizeclass);
           return nullptr;
         });
         return;
       }
 
+      remote_dealloc_and_post(target, p, sizeclass);
+    }
+
+    SNMALLOC_SLOW_PATH void
+    dealloc_not_large(RemoteAllocator* target, void* p, sizeclass_t sizeclass)
```

Comment:
I would have sooner this be closer to the callsite in `dealloc`, i.e. just before `dealloc_not_large_local`.